### DISCUSSION TITLE
Job owner problems in transformations after migrating to v8.0.39

### DISCUSSION THREAD

**Discussion by zhangxiaomei** [DIRACGrid/DIRAC#7526] [Q&A] [answered]
I didn't set shifterProxy for WorkflowTaskAgent-JUNO of TS [1]. In 8.0.24, the jobs created by WorkflowTaskAgent are using owner proxy of the transformation.  But after migrating to v8.0.39, all the jobs created from WorkflowTaskAgent are using a single shifter proxy with wrong VO group [1]. I am confused by that Any suggestions to debug that?
[1] 
![图片](https://github.com/DIRACGrid/DIRAC/assets/1860588/74196742-e4f0-49c2-b51e-820a0f233d10)

[2] submit a transformation by a  "cjiang" juno user, but jobs submitted with "zhangxm" bes user
@4000000065fbd0361fa52494.s:2024-03-21 06:07:56 UTC Transformation/WorkflowTaskAgent-JUNO/Transformation/WorkflowTaskAgent-JUNO [139749702289216] INFO: ShifterProxy: Tasks will be submitted with the credentials zhangxm:bes_user

**Comment by fstagni** [comment]
Can it be a consequence of https://github.com/DIRACGrid/DIRAC/pull/7113 ? @sfayer @marianne013

**Comment by andresailer** [comment, marked as answer]
My guess is that:

The issue is that there is a `/tmp/x509_u$(id -u dirac)` file on the server, and the WTA picks that up via

https://github.com/DIRACGrid/DIRAC/blob/26f60b621dcbc4c045196c510eb882a8478aed19/src/DIRAC/TransformationSystem/Agent/TaskManagerAgentBase.py#L106-L112

usually the shifter proxy will be created in some randomized x509 file and cross-talk is unlikely.
But then there is the fall back to
https://github.com/DIRACGrid/DIRAC/blob/26f60b621dcbc4c045196c510eb882a8478aed19/src/DIRAC/Core/Security/Locations.py#L19-L22

Probably we should make that fall back optional. I think this happened before already.

So for now delete the `/tmp/x509_u$(id -u dirac)` file from the sever.

**Reply by zhangxiaomei** [reply]
Yes, exactly. There is accidently /tmp/x509_u$(id -u dirac) in the server. I just realized that.

**Comment by marianne013** [comment]
If there's really a /tmp/x509_ *on the server, then usually all bets are off. I've left one accidentally more than once to all kinds of odd effects :-(

**Reply by zhangxiaomei** [reply]
At the beginning I didn't realize that "ProductionManagement" property need to be added to the host in 8.0.39 and got a permission error.  I put a proxy there for testing and forgot to delete it. 
Also I take it for granted that only valid proxy in the server would cause problems and didn't know even the proxy file would cause problem too :(
